route/src/route_generator_worker.cpp

```diff
@@ -438,5 +438,91 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crosstrack_error_check(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current, carma_wm::TrackPos llt_track)
+{
+  if (!world_model_->getMap()) 
+  {
+    ROS_ERROR_STREAM("WMListener received a route before a map was available. Dropping route message.");
+    return true;
+  }
+
+   
+
+  auto route = lanelet::ConstLanelets();
+  for(auto id : route_msg_.route_path_lanelet_ids)
+  {
+    auto ll = world_model_->getMap()->laneletLayer.get(id);
+    route.push_back(ll);
+  }
+
+  bool out_of_llt_bounds = false;
+
+    lanelet::BasicPoint2d position;
+
+    position.x()= msg->pose.position.x;
+    position.y()= msg->pose.position.y;  
+
+    if (boost::geometry::distance(position, current.polygon2d()) > cross_track_dist) //Evaluate lanelet crosstrack distance from vehicle
+        {
+            cte_count++;
+
+            if(cte_count > 4) //If the distance exceeds the crosstrack distance a certain number of times, report that the route has been departed
+                return true;
+        }
+
+         if (boost::geometry::distance(position, current.polygon2d()) > llt_track.downtrack) //Evaluate lanelet downtrack distance
+        {
+            cte_count++;
+
+            if(cte_count > 4)
+                return true;
+        }
+
+  if(boost::geometry::within(position, current.polygon2d()) == false) //Determine whether or not the vehicle is in the lanelet polygon
+      {
+        out_of_llt_bounds = true;
+      }
+
+    auto following_llts = world_model_->getMapRoutingGraph()->following(current); //Get all subsequent lanelets from the map
+    lanelet::ConstLanelets filtered_llts;
+    for(auto i:following_llts)
+    {
+       for(auto j: route)
```

Comment:
If this is using the route variable I was looking at in a prior comment, this is all lanelets on the map to my understanding, so this whole for-loop wouldn't eliminate any lanelets?

Reply:
This loop was initially made under the assumption that there may be lanelets in the map that were not a part of the shortest path route. This should be unnecessary, since the operation is meant to analyze all of the lanelets and not just the shortest path. I'll remove it if that's the case.